### DISCUSSION TITLE
PostgreSQL v15 - "invalid ELF header" Error

### DISCUSSION THREAD

**Discussion by peterlang-p** [docker-library/postgres#1214] [General]
Hello everyone,
I'm currently facing a perplexing issue with my Paperless setup running in a Proxmox container using Docker-compose. Everything was running smoothly until yesterday when I encountered a disk space shortage. To address this, I shut down the instance, resized the disk, and rebooted it. However, since then, I've been unable to start Paperless.

The root of the problem appears to be related to PostgreSQL. Below is an excerpt from the failure log:

```
paperless-ngx_db_1  | /usr/lib/postgresql/15/bin/postgres: error while loading shared libraries: /lib/x86_64-linux-gnu/libpam.so.0: invalid ELF header
paperless-ngx_db_1  | no data was returned by command ""/usr/lib/postgresql/15/bin/postgres" -V"
paperless-ngx_db_1  | initdb: error: program "postgres" is needed by initdb but was not found in the same directory as "/usr/lib/postgresql/15/bin/initdb"
```

So I was trying to dig a bit into and was able to confirm, that somehow the PostgreSQL v15 seems to be the issue in that case. (Paperless-ngx uses v15 by default, and I've been using it until now without a problem, that's why I'm trying to get that working again)

Spinning up a simple setup produces the same error:

**Docker-Compose:**
```

services:
  db:
    image: postgres:15
    restart: always
    environment:
      POSTGRES_PASSWORD: example
    volumes:
      - ./pgdata:/var/lib/postgresql/data
```

**Log:**
```
db-1  | PostgreSQL Database directory appears to contain a database; Skipping initialization
db-1  | 
db-1  | postgres: error while loading shared libraries: /lib/x86_64-linux-gnu/libpam.so.0: invalid ELF header
db-1 exited with code 0
```

PostgreSQL version 15.3 on the other hand starts without a problem.

Unfortunately, I'm at a loss regarding how this issue occurred and how to rectify it. I'm reaching out to the community for any insights or suggestions you may have.

Your help would be greatly appreciated. Thank you in advance for your time and assistance!

```
Docker Compose version v2.25.0
Docker Engine - Version:  26.0.0
```

Best regards,
Peter

**Comment by yosifkit** [comment]
I'd guess that some of the docker images are corrupted or have partially missing content because it run out of space. I'd recommend deleting the images and re-pulling them.

**Reply by peterlang-p** [reply]
I also considered that approach, but unfortunately, it didn't help. Upon further investigation, I discovered that at the moment my only solution is to run `docker system prune`. Afterwards I can start the Postgres container without any issues. However, I'm searching for a more sensitive solution that preserves my data. It appears that docker system prune performs additional actions that my manual steps overlook.
